pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -232,6 +250,188 @@ func getMachineDefaulterOperation(platformStatus *osconfigv1.PlatformStatus) mac
+		Path:      pointer.StringPtr(DefaultMachineValidatingHookPath),
+	}
+	return admissionregistrationv1.ValidatingWebhook{
+		AdmissionReviewVersions: []string{"v1beta1"},
+		Name:                    "validation.machine.machine.openshift.io",
+		FailurePolicy:           &webhookFailurePolicy,
+		SideEffects:             &webhookSideEffects,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			Service: &machineServiceReference,
+		},
+		// We need to change the order of list element before extending the list with new values
+		// or the DeepDerivative won't recognize the change
+		Rules: []admissionregistrationv1.RuleWithOperations{
+			{
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{machine.GroupName},
+					APIVersions: []string{SchemeGroupVersion.Version},
+					Resources:   []string{"machines"},
+				},
+				// We need to change the order of list element before extending the list with new values
+				// or the DeepDerivative won't recognize the change
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+					admissionregistrationv1.Update,
+				},
+			},
+		},
+	}
+}
+
+// MachineSetValidatingWebhook returns validating webhooks for machineSet to populate the configuration
+func MachineSetValidatingWebhook() admissionregistrationv1.ValidatingWebhook {
+	machinesetServiceReference := admissionregistrationv1.ServiceReference{
+		Namespace: defaultWebhookServiceNamespace,
+		Name:      webhookName,
+		Path:      pointer.StringPtr(DefaultMachineSetValidatingHookPath),
+	}
+	return admissionregistrationv1.ValidatingWebhook{
+		AdmissionReviewVersions: []string{"v1beta1"},
+		Name:                    "validation.machineset.machine.openshift.io",
+		FailurePolicy:           &webhookFailurePolicy,
+		SideEffects:             &webhookSideEffects,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			Service: &machinesetServiceReference,
+		},
+		// We need to change the order of list element before extending the list with new values
+		// or the DeepDerivative won't recognize the change
+		Rules: []admissionregistrationv1.RuleWithOperations{
+			{
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{machine.GroupName},
+					APIVersions: []string{SchemeGroupVersion.Version},
+					Resources:   []string{"machinesets"},
+				},
+				// We need to change the order of list element before extending the list with new values
+				// or the DeepDerivative won't recognize the change
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+					admissionregistrationv1.Update,
+				},
+			},
+		},
+	}
+}
+
+// NewMutatingWebhookConfiguration creates a mutating webhook configuration with configured Machine and MachineSet webhooks
+func NewMutatingWebhookConfiguration() *admissionregistrationv1.MutatingWebhookConfiguration {
+	mutatingWebhook := &admissionregistrationv1.MutatingWebhookConfiguration{
```

Comment:
to be consistent with the naming conventions you are following this should be `mutatingWebhookConfiguration :=`?